### PR TITLE
Milestone 5: Friendships v1

### PR DESCRIPTION
**Pull request by Laguna1**
A 'Friendship' model with associations and all requested features for friendships have been created.
Added tests.

### CODE REVIEW

**Review by adaorachi — CHANGES_REQUESTED** [review]
### Hi @Laguna1 and @chinweokwu :wave: 
Good job on the completion of the first draft of this milestone. The friendship/user model associations were well implemented. 

However, there are some changes you need to make which includes ensuring that these association features are implemented in the views and controller files and a user can see these features when the app is opened in the browser.
Kindly read my comments below.

---

### STATUS: CHANGES REQUIRED :recycle: 

---

Currently, you have no views nor controllers files for the friendship features. So in your views and controllers, kindly update them to meet the following requirements and features -

- [ ] There should be an invite button where a user can click to send a friendship invitation to friends. Here is a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/send_invitation.png) for clarification
- [ ] The button mentioned above should be visible and clearly seen on the users list page and on single-user page. Here's a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) in that regard. 
- [ ] A user can see the friendships requests made to him/her by the other users. Here is a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from pathwright.
- [ ] A user has to accept a friendship request to start a friendship with other users.
- [ ] A reject friendship feature is also present in your app in which a user can reject a friendship request send by another user.
- [ ] In your app, the posts timeline should include all posts written by the logged-in user and his/her friends. Here's a [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from pathwright for clarification.

---

### After implementing these changes, you can request for another code review

---

Should you have any question, please do reach out to me on Slack with the username @adaorachi

Happy Coding :relaxed: 

MaryAnn Chukwuka

**Comment by Laguna1** [comment]
Some changes have been made.

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
+require 'capybara/rspec'
+ENV['RAILS_ENV'] ||= 'test'
```
Comment (kubilaycaglayan) [inline spec/rails_helper.rb]: :bulb: [OPTIONAL] These two lines of `require` statements **caused issues** on my side. I had to delete them in order to run the tests. Consider deleting them, or please instruct users about how to run tests 🙏

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
Hello Team 👋
Status: Changes Required ♻️

Great job so far! 👏👏👏
Thanks for making the changes from the previous review 😊
Let's cover a few details to make this project better ⏬

### [REQUIRED] Behavioral Features
- [x] There is a button to invite the user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

💡 Currently, we have it on the single user page, but we don't have it on the users list page. Please add it.

- [x] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

💡 Currently, the user can only accept the invitations. Please also add the "reject invitation" feature. 😇

- [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

💡 Currently, in the posts timeline, we have all the posts from all the users. Please add a filter feature to show only the posts from logged-in user and their friends. 🙏

## That's it! After making the changes, you can request another code review 😊

With all my respect.🙏

**Pull request closed without merging**